integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/mutation/CarbonProjectForUpdateCommand.scala

```diff
@@ -60,6 +60,13 @@ private[sql] case class CarbonProjectForUpdateCommand(
       return Seq.empty
     }
     val carbonTable = CarbonEnv.getCarbonTable(databaseNameOp, tableName)(sparkSession)
+    if (carbonTable.getPartitionInfo != null &&
+      (carbonTable.getPartitionInfo.getPartitionType == PartitionType.RANGE ||
+        carbonTable.getPartitionInfo.getPartitionType == PartitionType.HASH ||
+        carbonTable.getPartitionInfo.getPartitionType == PartitionType.LIST)) {
```

Comment:
carbonTable.getPartitionInfo.getPartitionType.ordinal  != 4

Reply:
i think ordinal is not a good choice, because others may insert the enum and will not consider the ordinal,  right?

Reply:
why not check: RANGE_INTERVAL

Reply:
@QiangCai  the document doesn't describe the RANGE_INTERVAL and there isn't a test case to test RANGE_INTERVAL, so i didn't find the scene of RANGE_INTERVAL and no one use this parameter
was it discarded?   need it to add this check ?